opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -85,40 +95,50 @@ public void extendTypeMapping(Map<String, OpenSearchDataType> typeMapping) {
 
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-  private final Map<ExprType, Function<Content, ExprValue>> typeActionMap =
-      new ImmutableMap.Builder<ExprType, Function<Content, ExprValue>>()
+  private static DateTimeFormatterBuilder dateTimeFormatterbuilder =
+      new DateTimeFormatterBuilder()
+          .appendOptional(SQL_LITERAL_DATE_TIME_FORMAT)
+          .appendOptional(STRICT_DATE_OPTIONAL_TIME_FORMATTER)
+          .appendOptional(STRICT_HOUR_MINUTE_SECOND_FORMATTER);
```

Comment:
I think this should be fine? This is only slightly modified from how it was implemented [here](https://github.com/Bit-Quill/opensearch-project-sql/pull/169/files) (I modified it as a potential way to try and use user-defined formats). However, this might not even be needed anymore. `constructTimestamp` is only called in one place, if the call to `parseTimestampString` fails to parse the value, so probably the call to to `constructTimestamp` isn't even needed. I will try to rework.